src/rotation.zig

```diff
@@ -226,6 +254,21 @@ pub const RotationModes = struct {
 			return blocks.meshes.modelIndexStart(block) + @min(block.data, 3);
 		}
 
+		fn rotateZ(data: u16, angle: Degrees) u16 {
+			comptime var rotationTable: [4][4]u8 = undefined;
+			comptime for(0..4) |i| {
+				rotationTable[0][i] = i + 2;
+			};
+			comptime for(1..4) |a| {
+				for(0..4) |i| {
+					const neighbor: Neighbor = @enumFromInt(rotationTable[a - 1][i]);
+					rotationTable[a][i] = neighbor.rotateZ().toInt();
+				}
+			};
+			if(data >= 4) return 0;
+			return rotationTable[@intFromEnum(angle)][data] - 2;
```

Comment:
Could you precompute the -2 into the table?

Reply:
Yes, but you have to check if I didn't mess it up.